extensions/common/metadata_object.cc

```diff
@@ -0,0 +1,227 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "extensions/common/metadata_object.h"
+
+#include "flatbuffers/flatbuffers.h"
+#include "source/common/common/hash.h"
+
+namespace Istio {
+namespace Common {
+
+WorkloadMetadataObject WorkloadMetadataObject::fromBaggage(
+    absl::string_view baggage_header_value) {
+  // TODO: check for well-formed-ness of the baggage string: duplication,
+  // inconsistency
+  absl::string_view instance;
+  absl::string_view cluster;
+  absl::string_view workload;
+  absl::string_view namespace_name;
+  absl::string_view canonical_name;
+  absl::string_view canonical_revision;
+  absl::string_view app_name;
+  absl::string_view app_version;
+  WorkloadType workload_type = WorkloadType::KUBERNETES_POD;
+  const std::vector<std::string> empty;
+
+  std::vector<absl::string_view> properties =
+      absl::StrSplit(baggage_header_value, ',');
+  for (absl::string_view property : properties) {
+    std::pair<absl::string_view, absl::string_view> parts =
+        absl::StrSplit(property, "=");
+    if (parts.first == "k8s.namespace.name") {
```

Comment:
change these `if-else` to `switch-case`, WDYT?

Reply:
You can't switch on strings :) But yes, we can tokenize here with enums, will do that.